marine/marine.py

```diff
@@ -166,22 +174,42 @@ def filter_and_parse(
 
     def validate_bpf(
         self, bpf: str, encapsulation_type: int = encap_consts.ENCAP_ETHERNET
-    ) -> bool:
+    ) -> MarineValidationResult:
         bpf = bpf.encode("utf-8")
-        return bool(self._marine.validate_bpf(bpf, encapsulation_type))
+        err_msg = pointer(POINTER(c_char)())
+        valid = bool(self._marine.validate_bpf(bpf, encapsulation_type, err_msg))
+        if err_msg.contents:
+            err_msg_value = string_at(err_msg.contents)
+            self._marine.marine_free_err_msg(err_msg.contents)
+            return MarineValidationResult(valid, [err_msg_value.decode("utf-8")])
+        return MarineValidationResult(valid, [])
 
-    def validate_display_filter(self, display_filter: str) -> bool:
+    def validate_display_filter(self, display_filter: str) -> MarineValidationResult:
         display_filter = display_filter.encode("utf-8")
-        return bool(self._marine.validate_display_filter(display_filter))
+        err_msg = pointer(POINTER(c_char)())
+        valid = bool(self._marine.validate_display_filter(display_filter, err_msg))
+        if err_msg.contents:
```

Comment:
Maybe wrap those few lines by a functions that returns a tuple?

Reply:
I decided to encapsulate only the handling of `err_msg`